include/ocpp/v201/charge_point.hpp

```diff
@@ -648,6 +676,21 @@ class ChargePoint : ocpp::ChargingStationBase {
     DataTransferResponse data_transfer_req(const CiString<255>& vendorId, const std::optional<CiString<50>>& messageId,
                                            const std::optional<std::string>& data);
 
+    /// \brief Switches the operative status of the CS, an EVSE, or a connector, and recomputes effective statuses
+    /// \param evse_id: The ID of the EVSE, empty if the CS is addressed
+    /// \param connector_id: The ID of the connector, empty if an EVSE or the CS is addressed
+    /// \param new_status: The new operative status to switch to
+    /// \param persist: True if the updated state should be persisted in the database
+    void set_operative_status(std::optional<int32_t> evse_id, std::optional<int32_t> connector_id,
+                              OperationalStatusEnum new_status, bool persist);
+
+    /// \brief Explicitly trigger the change_effective_availability_callback for each component (done on boot)
+    void trigger_change_effective_availability_callback();
+
+    /// \brief Call the change_effective_availability_callback and the status_notification_callback if state changed
+    /// Also triggers the corresponding callback on EVSEs and connectors
+    void trigger_callbacks_if_effective_state_changed();
```

Comment:
are those two really necessary?
(The first I would think is triggered at init anyway since I would expect the state changing from "nothing" to "something" should trigger all callbacks; although I would understand that this could be a timing issue?;

The second I don't quite get - shouldn't the callbacks be triggered always when the state changes?

Reply:
Removed - the component state manager is now responsible for calling the callbacks as well